File: Backend/TeslaACDC.Data/Models/Autor.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace TeslaACDC.Data.Models;

public class Autor : BaseEntity<int>
{
    public string Nombre { get; set; } = string.Empty;
    public string Apellidos { get; set; } = string.Empty;
    public string Seudonimo { get; set; } = string.Empty;
    public string? Foto { get; set; }
    public DateOnly? FechaNacimiento { get; set; }
    public DateOnly? FechaMuerte { get; set; }
    public string Pais { get; set; } = string.Empty;
    public string Nacionalidad { get; set; } = string.Empty;
    public bool EstaVivo { get; set; }
    public string Biografia { get; set; } = string.Empty;
    public string Idioma { get; set; }

    public ICollection<LibroAutor> LibroAutores { get; set; } = new List<LibroAutor>();
    public ICollection<AudiolibroAutor> AudiolibroAutores { get; set; } = new List<AudiolibroAutor>();
    public ICollection<AutorGenero> AutorGeneros { get; set; } = new List<AutorGenero>();



}

File: Frontend/app/models/Autor.ts
export interface AutorResumen {
  id: number;
  nombre: string;
  apellidos: string;
  nacionalidad: string;
  foto: string;
  generos: string;
}
export interface Autor extends AutorResumen {
  pseudonimo: string;
  fechaNacimiento: Date;
  pais: string;
  estaVivo: boolean;
  biografia: string;
  idioma: string;
}

export interface AutorResponse {
  message: string;
  statusCode: number;
  totalElements: number;
  responseElements: Autor[];
}

export interface AutorCrear {
  id: number;
  nombre: string;
  apellido: string;
  generoId: string;
  foto: string;
  nacionalidad: string;
  biografia: string;
  idioma: string;
}
